lib/spack/docs/build_systems/inteloneapipackage.rst

```diff
@@ -0,0 +1,265 @@
+====================================
+
+Spack can also use oneAPI tools that are manually installed with
+`Intel Installers`_.  The procedures for configuring Spack to use
+external compilers and libraries are different.
+
+Compilers
+---------
+
+To use the compilers, add some information about the installation to
+``compilers.yaml``. For most users, it is sufficient to do::
+
+  spack compiler add /opt/intel/oneapi/compiler/latest/linux/bin/intel64
+  spack compiler add /opt/intel/oneapi/compiler/latest/linux/bin
+
+Adapt the paths above if you did not install the tools in the default
+location. After adding the compilers, using them in Spack will be
+exactly the same as if you had installed the
+``intel-oneapi-compilers`` package.  Another option is to manually add
+the configuration to ``compilers.yaml`` as described in :ref:`Compiler
+configuration <compiler-config>`.
+
+
+Libaries
+--------
+
+Configure external librar packages by editing ``packages.yml``,
+following the Spack documentation under :ref:`External Packages
+<sec-external-packages>`.
+
+Similar to ``compilers.yaml``, the ``packages.yaml`` files define a
+package external to Spack in terms of a Spack spec and resolve each
+such spec via veither the ``paths`` or ``modules`` tokens to a
+specific pre-installed package version on the system.  Since Intel
+tools generally need environment variables to interoperate, which
+cannot be conveyed in a mere ``paths`` specification, the ``modules``
+token will be more sensible to use. It resolves the Spack-side spec to
+a modulefile generated and managed outside of Spack's purview, which
+Spack will load internally and transiently when the corresponding spec
+is called upon to compile client packages.
+
+Unlike compilers, Spack does not offer a command to generate an
+entirely new ``packages.yaml`` entry.  You must create new entries
+yourself in a text editor, though the command ``spack config
+[--scope=...] edit packages`` can help with selecting the proper file.
+See section :ref:`Configuration Scopes <configuration-scopes>` for an
+explanation about the different files and section :ref:`Build
+customization <build-settings>` for specifics and examples for
+``packages.yaml`` files.
+
+
+Using oneAPI Tools Installed by Spack
+=====================================
+
+Spack can be a convenient way to install and configure compilers and
+libaries, even if you do not intend to build a Spack package. If you
+want to build a Makefile project using Spack-installed oneAPI compilers,
+then use spack to configure your environment::
+
+  spack load intel-oneapi-compilers
+
+And then you can build with::
+
+  CXX=icpx make
+
+You can also use Spack-installed libraries. For example::
+
+  spack load intel-oneapi-mkl
+
+Will update your environment CPATH, LIBRARY_PATH, and other
+environment variables for building an application with MKL.
+
+
+Debug notes
+===========
+
+
+* You can trigger a wall of additional diagnostics using Spack options, e.g.:
```

Comment:
This section is helpful but I would expect that in large part we should have covered it elsewhere. That being said perhaps you spoke with people who thought this should be available "up front" and that it was hard to dig into the docs to find this info. I think this can stay if that's why you included it here, and perhaps we should file an issue about the organization of the docs to help people find this more easily.

Reply:
Removed

This whole document was a cut/paste/edit of the intelpackage.rst
I am happy to remove more.